track.py

```diff
@@ -1,153 +1,87 @@
-            p1 = self.curr
-        return (
-            self._getPoint(p1)["geometry"]["coordinates"][1],
-            self._getPoint(p1)["geometry"]["coordinates"][0]
-        )
-
-    # private functions you shouldn't have to use
-    def _getPoint(self, p1=None):
-        if(p1 == None):
-            p1 = self.curr
-        return self.track["features"][self.name[p1]]
-
-    def _getElevation(self, p1=None):
-        if(p1 == None):
-            p1 = self.curr
-        return self._getPoint(p1)["properties"]["elevation"]
-
-    def _getDirectDistance(self, p1, p2):
-        lat1, lon1 = self.getCoords(p1)
-        lat2, lon2 = self.getCoords(p2)
-        x = 288200 * (lon2 - lon1)
-        y = 364000 * (lat2 - lat1)
-        d = math.sqrt(x * x + y * y)
-        if("radius" in self._getPoint(p1)["properties"]):
-            r = self.getRadius(p1)
-            return 2*r*math.asin(d/(2*r))
-        return d
-    
-def test():
-    t = Track("trackDynamic")
-    t.setForks("left", "left", "left")
-
-    while t.getNext(t.getCurr()) != "S0":
-        print(t.getCurr())
-        t.goNext()
-    print(t.getCurr())
-    t.goNext()
-
-    print("shortest track length: " + str(t.getDistance("S0", "S0"))) #
-    t.pitNext()
-    print("above length with pit lap: " + str(t.getDistance("S0", "S0"))) # this inherently 2 laps to make it to the finish line after a pit
-    t.setForks("right", "right", "right")
-    print("longest track length: " + str(t.getDistance("S0", "S0")/5280))
-    # t.generateMinimap()
-    # t.interpolateMinimap()
-
-test()
\ No newline at end of file
+class Track:
+  # PARAMS:
+  #   trackFile: the name of the track file to be used
+  def __init__(self, trackFile, friction=.01) -> None:
+    points = []
+    with open(trackFile) as f:
+      geojson = json.load(f)
+      features = geojson["features"]
+      for feature in features:
+          geometry = feature["geometry"]
+          coordinates = geometry["coordinates"]
+          properties = feature["properties"]
+
+          points.append([*coordinates, properties["elevation"]])
+          self.coords = coordinates
+
+
+    # Determine aspect ratio for scaling
+    minLat, maxLat, minLon, maxLon = 90, -90, 180, -180
+    for point in points:
+        if point[0] < minLon:
+            minLon = point[0]
+        if point[0] > maxLon:
+            maxLon = point[0]
+        if point[1] < minLat:
+            minLat = point[1]
+        if point[1] > maxLat:
+            maxLat = point[1]
+    aspectRatio = (maxLat - minLat)/(maxLon - minLon)
+
+    # Scale points and convert to km
+    for point in points:
+        point[0] = (point[0] - minLon)/(maxLon - minLon)
+        point[1] = (point[1] - minLat)/(maxLat - minLat)
+    points = np.array(points)
+    points[:,1] = points[:,1]*aspectRatio
+    points = points*.65 #rough conversion to km from lat/lon
+    # ^- conversion not accurate, maybe an issue the points. Should be .4 multiple, but the track would too short then
```

Comment:
do we need to make this accurate..?

Reply:
I think this conversion is good enough for us without losing too much accuracy